src/RA_LeaderboardPopup.cpp

```diff
@@ -290,14 +290,20 @@ void LeaderboardPopup::Render(HDC hDC, RECT& rcDest)
                         SetTextColor(hDC, COL_TEXT_HIGHLIGHT);
                     }
 
-                    ra::tstring str;
+                    
                     {
-                        std::basic_ostringstream<TCHAR> oss;
-                        oss << " " << lbInfo.m_nRank << " " << lbInfo.m_sUsername << " ";
-                        str = oss.str();
+                        std::string str{ "" };
+                        const auto nCharsWritten
+                        {
+                            std::snprintf(str.data(), BUFSIZ, " %u %s ", lbInfo.m_nRank,
```

Comment:
You need to `str.reserve(BUFSIZ)`. The initial capacity of a `std::string` is much smaller than `BUFSIZ`.

Reply:
Pretty sure you also need to do a `str.resize(nCharsWritten)` afterwards to update the length.

Reply:
Sure.

Reply:
The resize thing didn't work, but doing a "self" assign did.
`str = str.c_str();`

Reply:
Here's an example of using snprintf with a string buffer, in a function called "StringPrintf": https://github.com/RetroAchievements/RAIntegration/blob/master/src/RA_StringUtils.cpp#L129

Maybe you should just call that?

Reply:
Seems like a good approach

Reply:
I'll still use your suggestion but note it is slower than before (as well as not safe with the C varargs).
average over 1 million times.
`with ra::StringPrintf: 13.280193 microseconds`
instead of the 3.xxxxx microseconds.
It's not that big a deal, 1 millisecond = 1000 microseconds.